dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp

```diff
@@ -970,6 +944,18 @@ void CudaIcoCodeGen::generateAllAPIVerifyFunctions(
       dawn_unreachable("invalid location type");
     }
   };
+  auto getStride = [&](const sir::UnstructuredFieldDimension& unstrDims) -> std::string {
+    auto lt = unstrDims.getDenseLocationType();
+    std::string denseSize =
+        locToDenseSizeStringGpuMesh(lt, codeGenOptions.UnstrPadding, /*addParens*/ false);
+    std::string stride = "mesh." + denseSize;
+    if(unstrDims.isSparse()) {
+      std::string sparseSize = chainToSparseSizeString(unstrDims.getIterSpace());
+      stride = "(mesh." + denseSize + ") * " + "dawn_generated::cuda_ico::" + wrapperName +
+               "::" + sparseSize;
+    }
+    return stride;
+  };
```

Comment:
This stride is wrong: dense and sparse are not always adjacent dimensions, in a sparse full field they are interleaved by the vertical dimension. See assumption 1 of report.
The reason why `verify_dense_full_field()` is named as such instead of just `verify()` is related to this. The implementation of `verify_sparse_full_field()` would need to differ for the above reason (and for other reasons also).